lib/CloudinaryPlugin.js

```diff
@@ -0,0 +1,333 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2019 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const Cloudinary = require('cloudinary').v2;
+
+class CloudinaryPlugin {
+
+  constructor() {
+    this.context = null;
+  }
+
+  init(config, context) {
+    this.context = context;
+
+    this.controllers = {
+      assetsController: {
+        search: 'search',
+        rename: 'rename',
+        destroy: 'destroy',
+        transform: 'transform'
+      },
+
+      tagsController: {
+        add_tag: 'add_tag',
+        remove_tag: 'remove_tag',
+        remove_all_tags: 'remove_all_tags',
+        replace_tag: 'replace_tag'
+      },
+    };
+
+    this.routes = [
+      {
+        verb: 'get',
+        url: '/assets/:expression',
+        controller: 'assetsController',
+        action: 'search'
+      },
+      {
+        verb: 'post',
+        url: '/assets/transform',
+        controller: 'assetsController',
+        action: 'transform'
+      },
+      {
+        verb: 'put',
+        url: '/assets',
+        controller: 'assetsController',
+        action: 'rename'
+      },
+      {
+        verb: 'delete',
+        url: '/assets',
```

Comment:
same as for transform.
Further, even if not explicitly forbidden, `DELETE` requests usually do not have a body.

Reply:
Well i have the same problem as [above](https://github.com/kuzzleio/kuzzle-plugin-cloudinary/pull/2#discussion_r296140442). Isn't it better to keep the public_id in the request body, even if it's not very conventional ?

Reply:
Hum... Even if not very RESTful, we can use a body parameter for a `POST` request like for `transform` and you made a point about the slashes being accepted by cloudinary.

The `DELETE` case is much more problematic to me. 
From [this post](https://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request), I got at least three blocking cases:

1. Android clients cannot send a `DELETE` query including a body.
2. GCE load balancer will respond with a 400 error in this case
3. The OPENAPI v3 explicitely forbids a body for `DELETE` requests

As far as I can see, this leaves us with two choices:

- either we stick to a RESTful format and include the parameter in the url, implying we force the clients to URL-encore it
- or we use another http verb (`POST`) for such requests to be able to keep the parameter within the body.

I guess this should be discussed internally. I opened a discussion about it.